ldpd/ldpd.c

```diff
@@ -140,6 +140,7 @@ static struct option longopts[] =
 	{ "ctl_socket",  required_argument, NULL, OPTION_CTLSOCK},
 	{ "user",        required_argument, NULL, 'u'},
 	{ "group",       required_argument, NULL, 'g'},
+	{ "instance",    required_argument, NULL, 'n'},
```

Comment:
Ok, this was suggested on the call but I think we don't need to do this in ldpd. With OSPF/EIGRP we might have multiple instances of a routing daemon running on the same VRF. For LDP this wouldn't make any sense (what would be the point?). If I got things right, you're running multiple instances of ldpd but they're all associated with different instances of zebra running on separate network namespaces. So I can't see any possible overlap here.

However I think you should keep the label manager aware of instance IDs because of daemons like ospfd (which will request labels as well once Olivier's SR code is merged).